superset/views/log/api.py

```diff
@@ -44,13 +56,82 @@ class LogRestApi(LogMixin, BaseSupersetModelRestApi):
         "referrer",
     ]
     show_columns = list_columns
+    apispec_parameter_schemas = {
+        "get_recent_activity_schema": get_recent_activity_schema,
+    }
+    openapi_spec_component_schemas = (
+        RecentActivityResponseSchema,
+        RecentActivitySchema,
+    )
 
     @staticmethod
     def is_enabled() -> bool:
         return app.config["FAB_ADD_SECURITY_VIEWS"] and app.config["SUPERSET_LOG_VIEW"]
 
-    @before_request
+    @before_request(only=["get_list", "get", "post"])
     def ensure_enabled(self) -> None:
         if not self.is_enabled():
             return self.response_404()
         return None
+
+    def get_user_activity_access_error(self, user_id: int) -> Optional[FlaskResponse]:
+        try:
+            security_manager.raise_for_user_activity_access(user_id)
+        except SupersetSecurityException as ex:
+            return self.response(403, message=ex.message)
+        return None
+
+    @expose("/recent_activity/<int:user_id>/", methods=["GET"])
```

Comment:
With `ENABLE_BROAD_ACTIVITY_ACCESS` enabled, users can view other users' profiles, which will fetch the those users' recent activity.  I don't think we can make this change without breaking this behavior, which seems to me like something that would have to wait for 3.0 (correct me if I'm wrong).

I can think of one alternative: we make this new endpoint only get the current logged in user activity, and have the frontend for the profile page call the deprecated endpoint until it's removed in 3.0.  Then in 3.0 the ability to view other users' profiles would be removed, and we can switch that page to use the new endpoint.

However, I think that only makes sense if `ENABLE_BROAD_ACTIVITY_ACCESS` will be removed in 3.0, what do you think?

Reply:
Is there any way for a normal user to view other user's profile? without tempering with the URL?

Since the frontend always calls the old and new endpoint with the current user login `ENABLE_BROAD_ACTIVITY_ACCESS` will not affect the frontend in anyway. 

Would not be a breaking change IMO, since the old endpoint is still there and still supports `ENABLE_BROAD_ACTIVITY_ACCESS` up until 3.0, this functionality is only leveraged by calling the endpoint directly.

Yet, let's keep `ENABLE_BROAD_ACTIVITY_ACCESS` for now, since this config key is tied to other functionality also, we prob should tackle this in a more integrated way.

Reply:
There is a way to view another user's profile without modifying the URL, on the dashboard and chart listviews the "Modified By" column will display a link to the user's profile who last modified the entity when `ENABLE_BROAD_ACTIVITY_ACCESS` is enabled:
![Screen Shot 2023-01-24 at 11 15 11 AM](https://user-images.githubusercontent.com/47196419/214387668-0730d3b7-6390-4f3c-9e80-86f0b59e13ba.png)

Then the frontend of the profile page will request that user's activity

Reply:
oh true, I remember that one!